WordPress/Classes/ViewRelated/System/Sidebar/SidebarViewController.swift

```diff
@@ -221,21 +222,47 @@ private struct SidebarProfileContainerView: View {
     }
 }
 
-private extension BlogListViewModel {
-    /// Returns the most recent sites and mixes and mixes in the rest of the sites
-    /// until the display limimt is reached.
+extension BlogListViewModel {
+    /// Returns a list of sites to display in the sidebar, ensuring that:
+    /// 1. The current site is always included
+    /// 2. The most recent sites are included up to the display limit
+    /// 3. The sites are sorted alphabetically
     var topSites: [BlogListSiteViewModel] {
-        var topSites = recentSites.prefix(SidebarView.displayedSiteLimit)
-        var encounteredIDs = Set(topSites.map(\.id))
-        for site in allSites where !encounteredIDs.contains(site.id) {
-            if topSites.count >= SidebarView.displayedSiteLimit {
+        var displaySites = [BlogListSiteViewModel]()
```

Comment:
Have you tried the steps from https://github.com/wordpress-mobile/WordPress-iOS/issues/23569? I vaguely remember this code, but I think we opened it just in case. I doubt it's a real issue. `recentSites` always has your most recently selected site as the first item of the array. It's impossible for it to not get displayed. I might be wrong – not entirely sure about this code.

Reply:
Yes, I followed the steps from the original issue, and I confirm I was able to replicate the issue, especially if you have multiple sites and you create or select a new one that starts with `W` or `Z`.

Screenshot:
![Simulator Screenshot - iPad Pro 13-inch (M4) - 2025-03-21 at 14 13 18](https://github.com/user-attachments/assets/951d7b14-2a2c-4a1f-ba99-e695af04779c)